x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -187,34 +193,34 @@
     private final TimeValue expirationDelay;
     private final TimeValue deleteInterval;
     private final Client client;
-    private final SecurityIndexManager securityIndex;
+    private final SecurityIndexManager securityMainIndex;
+    private final SecurityIndexManager securityTokensIndex;
     private final ExpiredTokenRemover expiredTokenRemover;
     private final boolean enabled;
     private volatile TokenKeys keyCache;
     private volatile long lastExpirationRunMs;
     private final AtomicLong createdTimeStamps = new AtomicLong(-1);
+    // for testing only
+    private final AtomicReference<Version> versionCompatibilityOverride = new AtomicReference<>();
```

Comment:
I know 😢 but I wish to change this in an `IntegTest` (`TokenAuthIntegTest`) because I need true indices created, not mocked clients. I will search for a better way, suggestions are welcome!

Reply:
Some of the tests I had in mind **require** the rolling upgrade setup because old node versions should be able to use/decode tokens. I will do most of the testing there.